packages/x-components/src/x-modules/tagging/store/actions/track-tagging.action.ts

```diff
@@ -0,0 +1,44 @@
+import { TaggingInfo } from '../../../../../../search-types';
+import { XPlugin } from '../../../../plugins/x-plugin';
+import { DefaultSessionService } from '../../service';
+import { TaggingXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link TaggingActions.track}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ * @param taggingInfo - The information of the event to track.
+ *
+ * @public
+ */
+export const track: TaggingXStoreModule['actions']['track'] = ({ state }, taggingInfo) => {
+  const { consent } = state;
+  const tagging = Array.isArray(taggingInfo) ? taggingInfo : [taggingInfo];
+  const sessionId = getSessionId(consent);
+
+  tagging.forEach(({ url, params }: TaggingInfo) => {
+    if (sessionId !== undefined) {
+      Object.assign(params, {
```

Comment:
With this `assign` you are mutating the parameter you receive, which isn't very safe, you don't know where this parameter comes from. But moreover, in our case it comes from the search module store, so you are mutating the search module state from this line, which is a very bad idea. Clone the object like you were doing before.

If you like one liners you can do it like this:

```ts
XPlugin.adapter.track({
      url,
      params: {
        ...params,
        ...(session && { session })
      }
    });
```